include/radproto/socket.h

```diff
@@ -13,11 +13,11 @@ namespace RadProto
     {
         public:
             Socket(boost::asio::io_service& io_service, const std::string& secret, uint16_t port);
-            void asyncReceive(const std::function<void(const boost::system::error_code&, const std::optional<Packet>&)>& callback);
-            void asyncSend(const Packet& response, const std::function<void(const boost::system::error_code&)>& callback);
+            void asyncReceive(const std::function<void(const boost::system::error_code&, const std::optional<Packet>&, const boost::asio::ip::udp::endpoint& source)>& callback);
```

Comment:
You do not need 'source' here, it's a type signature, not a function declaration.